$NetBSD: patch-src_Makefile,v 1.1 2019/06/30 21:41:46 nia Exp $

- Do not override CFLAGS/CXXFLAGS.
- Use libenet from pkgsrc.
- Use LDFLAGS instead of CXXFLAGS for linking.

--- src/Makefile.orig	2012-08-28 05:30:07.000000000 +0000
+++ src/Makefile
@@ -1,10 +1,9 @@
-CXXFLAGS= -O3 -fomit-frame-pointer
-override CXXFLAGS+= -Wall -fsigned-char -fno-exceptions -fno-rtti
+CXXFLAGS+= -Wall -fsigned-char -fno-exceptions -fno-rtti
 
 PLATFORM= $(shell uname -s)
 PLATFORM_PREFIX= native
 
-INCLUDES= -Ishared -Iengine -Ifpsgame -Ienet/include
+INCLUDES= -Ishared -Iengine -Ifpsgame $$(pkg-config --cflags libenet)
 
 STRIP=
 ifeq (,$(findstring -g,$(CXXFLAGS)))
@@ -37,7 +36,7 @@ endif
 CLIENT_LIBS= -mwindows $(STD_LIBS) -L$(WINBIN) -L$(WINLIB) -lSDL -lSDL_image -lSDL_mixer -lzlib1 -lopengl32 -lenet -lws2_32 -lwinmm
 else	
 CLIENT_INCLUDES= $(INCLUDES) -I/usr/X11R6/include `sdl-config --cflags`
-CLIENT_LIBS= -Lenet/.libs -lenet -L/usr/X11R6/lib -lX11 `sdl-config --libs` -lSDL_image -lSDL_mixer -lz -lGL
+CLIENT_LIBS= $$(pkg-config --libs libenet) -L/usr/X11R6/lib -lX11 `sdl-config --libs` -lSDL_image -lSDL_mixer -lz -lGL
 endif
 ifeq ($(PLATFORM),Linux)
 CLIENT_LIBS+= -lrt
@@ -106,7 +105,7 @@ SERVER_LIBS= -mwindows $(STD_LIBS) -L$(W
 MASTER_LIBS= $(STD_LIBS) -L$(WINBIN) -L$(WINLIB) -lzlib1 -lenet -lws2_32 -lwinmm
 else
 SERVER_INCLUDES= -DSTANDALONE $(INCLUDES)
-SERVER_LIBS= -Lenet/.libs -lenet -lz
+SERVER_LIBS= $$(pkg-config --libs libenet) -lz
 MASTER_LIBS= $(SERVER_LIBS)
 endif
 SERVER_OBJS= \
@@ -176,14 +175,14 @@ master: $(MASTER_OBJS)
 
 install: all
 else
-client:	libenet $(CLIENT_OBJS)
-	$(CXX) $(CXXFLAGS) -o sauer_client $(CLIENT_OBJS) $(CLIENT_LIBS)
+client:	$(CLIENT_OBJS)
+	$(CXX) $(LDFLAGS) -o sauer_client $(CLIENT_OBJS) $(CLIENT_LIBS)
 
-server:	libenet $(SERVER_OBJS)
-	$(CXX) $(CXXFLAGS) -o sauer_server $(SERVER_OBJS) $(SERVER_LIBS)  
+server:	$(SERVER_OBJS)
+	$(CXX) $(LDFLAGS) -o sauer_server $(SERVER_OBJS) $(SERVER_LIBS)  
 	
-master: libenet $(MASTER_OBJS)
-	$(CXX) $(CXXFLAGS) -o sauer_master $(MASTER_OBJS) $(MASTER_LIBS)  
+master: $(MASTER_OBJS)
+	$(CXX) $(LDFLAGS) -o sauer_master $(MASTER_OBJS) $(MASTER_LIBS)  
 
 shared/cube2font.o: shared/cube2font.c
 	$(CXX) $(CXXFLAGS) -c -o $@ $< `freetype-config --cflags`
